src/plugins/auto_batch/src/sync_infer_request.cpp

```diff
@@ -5,345 +5,340 @@
 ///////////////////////////////////////////////////////////////////////////////////////////////////
 #include "sync_infer_request.hpp"
 
+#include "openvino/core/type/element_type_traits.hpp"
+#include "transformations/utils/utils.hpp"
+
 namespace ov {
 namespace autobatch_plugin {
 
-template <InferenceEngine::Precision::ePrecision precision>
-InferenceEngine::Blob::Ptr create_shared_blob_on_top_of_batched_blob(InferenceEngine::Blob::Ptr batched_blob,
-                                                                     std::string name,
-                                                                     const std::set<std::string>& batched_names,
-                                                                     size_t batch_id,
-                                                                     size_t batch_num) {
-    typedef typename InferenceEngine::PrecisionTrait<precision>::value_type TYPE;
-    typedef typename std::add_pointer<TYPE>::type TYPEPTR;
-    auto ptr = batched_blob->buffer().as<TYPEPTR>();
-    auto sizePerBatch = batched_blob->size() / batch_num;
-    InferenceEngine::SizeVector dims = batched_blob->getTensorDesc().getDims();
+template <ov::element::Type_t precision>
+ov::Tensor create_shared_tensor_on_batched_tensor(ov::Tensor batched_tensor,
+                                                  std::string name,
+                                                  const std::set<std::string>& batched_names,
+                                                  size_t batch_id,
+                                                  size_t batch_num) {
+    using T = ov::fundamental_type_for<precision>;
+    auto ptr = static_cast<T*>(batched_tensor.data());
```

Comment:
looks like we don't need template here, we can just use `ptr` as `void*`

Reply:
@ilya-lavrenov  try your suggestion will get some accuracy issue, such as:
![image](https://github.com/openvinotoolkit/openvino/assets/52686861/33f8895e-8864-4425-926f-cfaa5e4b9227)

Reply:
because you also need to operate with batched_tensor.get_byte_size() instead of batched_tensor.get_size() in the code below.

Reply:
yes, you are right. I missed the change, with get_byte_size(), works now